### PR TITLE
Leaflet proj refactor

### PR DESCRIPTION
**Pull request by perliedman**
Leaflet 1.0 is out since a couple of weeks, and there's no reason Proj4Leaflet's `master` branch should use 0.7.

### CODE REVIEW

**Comment by theashyster** [comment]
I agree.

Although there are quite a lot of conflicts to fix and the documentation also would need to be updated to reflect the changes and reference the Leaflet 0.7 users to the commit before the merge of this pull request. Or should there be a release for the current code in master targeting Leaflet 0.7 and then a release of the new code in master targeting Leaflet 1.0?

This looks quite big since `master` and `leaflet-proj-refactor` branches were a bit out of sync and should be tested well.

**Comment by perliedman** [comment]
Yeah, the first commit in this PR is almost **three years old** :smiley_cat: 

I think it looks worse than it is, but yeah, this is going to require some review.

I forgotten when there was a 0.7 release last and if there's anything meaningful that has been done after that, but maybe @semone or @pthorin knows?

**Comment by semone** [comment]
I don't think there has been any major changes or releases for 0.7, not that i can remember, no new festures. 

I think we need to go through the docs since it has changed and uses gh-pages in master. I dont think its updated at all for 1.0. That is the only thing that i can recall

**Comment by theashyster** [comment]
Yeah, true, :)

I agree on that also, for me it looked that quite a lot of things could have been and probably should have been merged into master already, since they were not Leaflet 1.0 specific, but seems that those 2 branches just started to live life on their own at one point.

If we take a look at https://github.com/kartena/Proj4Leaflet/releases then the 0.7.2 release was a year ago and since then there has been 6 commits to the `master` branch, not actually doing a lot with code itself https://github.com/kartena/Proj4Leaflet/compare/0.7.2...master So it looks like the updated documentation can also reference release 0.7.2 for those who are still working with Leaflet 0.7.x, and it is as @semone says, nothing noteworthy there.

As for 1.0.0-beta.2 release there are 16 commits since last year https://github.com/kartena/Proj4Leaflet/compare/1.0.0-beta.2...leaflet-proj-refactor so after fixing the documentation, conflicts and merging this PR in a stable release 1.0.0 would be really awesome to get out.

I think the documentation in `leaflet-proj-refactor` is pretty much updated for Leaflet 1.0.

**Comment by bhaskarvk** [comment]
Any progress on this ?

**Comment by pthorin** [comment]
I am going to work on this now, hopefully I will get through it today.

File: examples/tms-tiled/index.html [hunk 1]
```diff
@@ -3,11 +3,19 @@
...
+  <body>
+    <div id="map"></div>
+    <script src="../../lib/leaflet/leaflet.js"></script>
+=======
     <link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet/v0.7.7/leaflet.css" />
   </head>
   <body>
     <div id="map"></div>
     <script src="http://cdn.leafletjs.com/leaflet/v0.7.7/leaflet.js"></script>
+>>>>>>> master
```
Comment (theashyster) [inline examples/tms-tiled/index.html]: @pthorin I saw that this is merged already, but looks like this was not fixed before merging.

Comment (pthorin) [inline examples/tms-tiled/index.html]: Oops, I missed pressing save on that one! I'm working on trying to get automated tests so can fix that soon as well. Thanks for noticing!

Comment (theashyster) [inline examples/tms-tiled/index.html]: :) Heh, good. Great job!

**Pull request merged**